### PR TITLE
refactor: remove circular dependencies

### PR DESCRIPTION
**Pull request by jinlee93**
[EDS-883]

### Summary:
- removes circular dependencies by 
  - importing components from component folder rather than `src/index`
  - moving Accordion subcomponents within Accordion file itself (could have been fixed via finagling exports/imports but this needed to be done anyways to follow subcomponent within main component file pattern)
- required to resolve [rollup warnings](https://github.com/chanzuckerberg/edu-design-system/pull/1555)

### Test Plan:

- [x] CI tests / new tests are not applicable
- [x] When cherry picked onto rollup branch and running `yarn build:js` no more circular deps warning

[EDS-883]: https://czi-tech.atlassian.net/browse/EDS-883?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: src/components/InlineNotification/InlineNotification.tsx [hunk 1]
```diff
@@ -1,6 +1,7 @@
 import clsx from 'clsx';
 import React from 'react';
-import { Icon, Text } from '../../';
+import Icon from '../Icon';
```
Comment (booc0mtaco) [inline src/components/InlineNotification/InlineNotification.tsx:3]: ++

File: src/components/ButtonDropdown/ButtonDropdown.tsx [hunk 2]
```diff
@@ -1,8 +1,8 @@
 import clsx from 'clsx';
 import type { ReactNode } from 'react';
 import React from 'react';
-import { DropdownMenu } from '../..';
 import type { ClickableStyleProps } from '../ClickableStyle';
+import DropdownMenu from '../DropdownMenu';
```
Comment (booc0mtaco) [inline src/components/ButtonDropdown/ButtonDropdown.tsx:5]: ++

**Comment by booc0mtaco** [comment]
@jinlee93 ahh, so you are seeing those random snapshot changes too. i guess chromatic must've changed?

**Review by booc0mtaco — APPROVED** [review]
Looks good

**Comment by jinlee93** [comment]
> @jinlee93 ahh, so you are seeing those random snapshot changes too. i guess chromatic must've changed?

Yea I saw it on the rollup pr first so I was like dang did the build really affect everything by 1px but I guess it's a chromatic thing

**Pull request merged**